### PR TITLE
Add support for HTTPS remote stations and MQTT support

### PR DESCRIPTION
**Pull request by salbahra**
Provides improved remote station support by adding HTTPS support. Also adds MQTT support.

### CODE REVIEW

File: www/js/main.js [hunk 1]
```diff
@@ -4289,25 +4295,39 @@ function showOptions( expandItem ) {
 					"</div>";
 		}
 
+		if ( typeof controller.settings.email !== "undefined" ) {
+			list += "<div class='ui-field-contain'>" +
+						"<label for='email'>" + _( "Email Notifs" ) +
```
Comment (salbahra) [inline www/js/main.js:4300]: Let's please spell this out `Notifications`

File: www/js/main.js [hunk 2]
```diff
@@ -4289,25 +4295,39 @@ function showOptions( expandItem ) {
...
+						"</label>" +
+						"<button data-mini='true' id='email' value='" + escapeJSON( controller.settings.email ) + "'>" +
+							_( "Tap to Configure" ) +
+						"</button>" +
+					"</div>";
+		}
+
 		if ( typeof controller.settings.ifkey !== "undefined" ) {
-			list += "<div class='ui-field-contain'><label for='ifkey'>" + _( "IFTTT Key" ) +
+			list += "<div class='ui-field-contain'><label for='ifkey'>" + _( "IFTTT Notifs" ) +
```
Comment (salbahra) [inline www/js/main.js:4313]: Please spell this out

File: www/js/main.js [hunk 3]
```diff
@@ -4289,25 +4295,39 @@ function showOptions( expandItem ) {
...
 				"<button data-helptext='" +
 					_( "To enable IFTTT, a Webhooks key is required which can be obtained from https://ifttt.com" ) +
 					"' class='help-icon btn-no-border ui-btn ui-icon-info ui-btn-icon-notext'></button>" +
-			"</label><input autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false' data-mini='true' type='text' id='ifkey' value='" + controller.settings.ifkey + "'>" +
+			"</label><input autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false' data-mini='true' type='text' id='ifkey' placeholder='IFTTT webhooks key' value='" + controller.settings.ifkey + "'>" +
 			"</div>";
 
-			list += "<div class='ui-field-contain'><label for='o49'>" + _( "IFTTT Events" ) +
+			list += "<div class='ui-field-contain'><label for='o49'>" + _( "Notif Events" ) +
```
Comment (salbahra) [inline www/js/main.js:4320]: Please spell this out

File: www/js/main.js [hunk 4]
```diff
@@ -4472,6 +4492,38 @@ function showOptions( expandItem ) {
 			}
 		} );
 
+	page.find( ".edit-loc" ).on( "click", function( e ) {
+		e.stopImmediatePropagation();
+
+		var popup = $( "<div data-role='popup' data-theme='a' id='locEntry'>" +
+			"<div data-role='header' data-theme='b'>" +
+				"<h1>" + _( "Enter GPS Coords" ) + "</h1>" +
```
Comment (salbahra) [inline www/js/main.js:4500]: Please spell this out

File: www/js/main.js [hunk 5]
```diff
@@ -4472,6 +4492,38 @@ function showOptions( expandItem ) {
...
+			"</div>" +
+			"<div class='ui-content'>" +
+				"<label id='loc-warning'>" + _( "" ) + "</label>" +
+				"<input class='loc-entry' type='text' id='loc-entry' data-mini='true' maxlength='64' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'"
+								 + " placeholder='" + _( "Enter GPS Coords" ) + "' value='" + ( controller.settings.loc.trim() === "''" ? _( "Not specified" ) : controller.settings.loc ) + "' required />" +
```
Comment (salbahra) [inline www/js/main.js:4505]: Please spell this out

File: www/js/main.js [hunk 6]
```diff
@@ -4472,6 +4492,38 @@ function showOptions( expandItem ) {
...
+			var input = popup.find( "#loc-entry").val();
+			const gpsre = /^[-+]?([1-8]?\d(\.\d+)?|90(\.0+)?),\s*[-+]?(180(\.0+)?|((1[0-7]\d)|([1-9]?\d))(\.\d+)?)$/;
+			if(gpsre.test(input)) {
+				page.find( "#loc" ).val( input ).removeClass( "green" ).find( "span" ).text( input );
+				page.find( "#o1" ).selectmenu( "disable" );
+				header.eq( 2 ).prop( "disabled", false );
+				page.find( ".submit" ).addClass( "hasChanges" );
+				popup.popup( "close" );
+			} else{
+				$("#loc-warning").css({'color':'maroon'}).text("Invalid GPS coords, try again");
```
Comment (salbahra) [inline www/js/main.js:4520]: Full word please

Comment (salbahra) [inline www/js/main.js:4520]: Don't we use a standard error class/color?

File: www/js/main.js [hunk 7]
```diff
@@ -4928,18 +4980,34 @@ function showOptions( expandItem ) {
 		openPopup( popup );
 	} );
 
+	function generateDefaultSubscribeTopic(){
+		let topic;
+		if(controller.settings.mac){
+			topic = controller.settings.mac;
+			topic = topic.replaceAll(":", "");
+			topic = "OS-" + topic;
+		}else{
```
Comment (salbahra) [inline www/js/main.js:4989]: Might fail lint rules for spaces

File: www/js/main.js [hunk 8]
```diff
@@ -4968,22 +5036,49 @@ function showOptions( expandItem ) {
...
+								"<input class='mqtt-input' type='text' id='pubt' data-mini='true' maxlength='24' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'" +
+									( options.en ? "" : "disabled='disabled'" ) + " placeholder='" + _( "publish topic" ) + "' value='" + options.pubt + "' required />" +
+							"</div>" : "" ) +
+							(largeSOPTSupport ?
+							"<div class='ui-block-a' style='width:40%'>" +
+								"<label for='subt' style='padding-top:10px'>" + _( "Subscribe Topic" ) + "</label>" +
+							"</div>" +
+							"<div class='ui-block-b' style='width:60%'>" +
+								"<input class='mqtt-input' type='text' id='subt' data-mini='true' maxlength='24' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'" +
+									( options.en ? "" : "disabled='disabled'" ) + " placeholder='" + _( "subscribe topic" ) + "' value='" + options.subt + "' required />" +
```
Comment (salbahra) [inline www/js/main.js:5063]: Should be capital?

File: www/js/main.js [hunk 9]
```diff
@@ -4968,22 +5036,49 @@ function showOptions( expandItem ) {
...
+								"<input class='mqtt-input' type='password' id='password' data-mini='true' maxlength='" + (largeSOPTSupport ? "100" : "32") + "' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'" +
 									( options.en ? "" : "disabled='disabled'" ) + " placeholder='" + _( "password (optional)" ) + "' value='" + options.pass + "' required />" +
 							"</div>" +
+							(largeSOPTSupport ?
+							"<div class='ui-block-a' style='width:40%'>" +
+								"<label for='pubt' style='padding-top:10px'>" + _( "Publish Topic" ) + "</label>" +
+							"</div>" +
+							"<div class='ui-block-b' style='width:60%'>" +
+								"<input class='mqtt-input' type='text' id='pubt' data-mini='true' maxlength='24' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'" +
+									( options.en ? "" : "disabled='disabled'" ) + " placeholder='" + _( "publish topic" ) + "' value='" + options.pubt + "' required />" +
```
Comment (salbahra) [inline www/js/main.js:5055]: Title casing for placeholders

File: www/js/main.js [hunk 10]
```diff
@@ -5016,6 +5113,103 @@ function showOptions( expandItem ) {
...
+								"<label for='port' style='padding-top:10px'>" + _( "Port" ) + "</label>" +
+							"</div>" +
+							"<div class='ui-block-b' style='width:60%'>" +
+								"<input class='email-input' type='number' id='port' data-mini='true' pattern='[0-9]*' min='0' max='65535'" +
+									( options.en ? "" : "disabled='disabled'" ) + " placeholder='465' value='" + options.port + "' required />" +
+							"</div>" +
+							"<div class='ui-block-a' style='width:40%'>" +
+								"<label for='username' style='padding-top:10px'>" + _( "Sender Email" ) + "</label>" +
+							"</div>" +
+							"<div class='ui-block-b' style='width:60%'>" +
```
Comment (salbahra) [inline www/js/main.js:5156]: Type email for this input

File: www/js/main.js [hunk 11]
```diff
@@ -5016,6 +5113,103 @@ function showOptions( expandItem ) {
...
+								"<input class='email-input' type='password' id='password' data-mini='true' maxlength='64' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'" +
+									( options.en ? "" : "disabled='disabled'" ) + " placeholder='" + _( "app password" ) + "' value='" + options.pass + "' required />" +
+							"</div>" +
+							"<div class='ui-block-a' style='width:40%'>" +
+								"<label for='recipient' style='padding-top:10px'>" + _( "Recipient Email" ) + "</label>" +
+							"</div>" +
+							"<div class='ui-block-b' style='width:60%'>" +
+								"<input class='email-input' type='text' id='recipient' data-mini='true' maxlength='64' autocomplete='off' autocorrect='off' autocapitalize='off' spellcheck='false'" +
```
Comment (salbahra) [inline www/js/main.js:5171]: Type email as well

File: www/js/main.js [hunk 12]
```diff
@@ -5640,19 +5849,22 @@ var showHome = ( function() {
...
 			// Station tab is initially set to disabled until we have refreshed station data from firmware
+			// Note: HTTPS and Remote OTC stations are supported at the same time with Email notification support
 			if ( Supported.special() ) {
 				select +=
 					"<div class='ui-bar-a ui-bar'>" + _( "Station Type" ) + ":</div>" +
 						"<select data-mini='true' id='hs'"  + ( Station.isSpecial( sid ) ? " class='ui-disabled'" : "" ) + ">" +
 							"<option data-hs='0' value='0'" + ( Station.isSpecial( sid ) ? "" : "selected" ) + ">" + _( "Standard" ) + "</option>" +
 							"<option data-hs='1' value='1'>" + _( "RF" ) + "</option>" +
-							"<option data-hs='2' value='2'>" + _( "Remote" ) + "</option>" +
+							"<option data-hs='2' value='2'>" + _( "Remote IP station" ) + "</option>" +
```
Comment (salbahra) [inline www/js/main.js:5859]: Remote Station (IP)

File: www/js/main.js (hunk 8 above)
Comment (rayshobby) [reply www/js/main.js:5063]: the other place holder values are non-capital, so im going to keep this one consistent with the others.

**Review by rayshobby — APPROVED** [review]
Looks good!

**Pull request merged**